streams/src/test/java/org/apache/kafka/streams/integration/RegexSourceIntegrationTest.java

```diff
@@ -198,9 +200,16 @@ public void testRegexRecordsAreProcessedAfterReassignment() throws Exception {
 
             final StreamsBuilder builder = new StreamsBuilder();
             final KStream<String, String> pattern1Stream = builder.stream(Pattern.compile("TEST-TOPIC-\\d"));
-            pattern1Stream.to(outputTopic, Produced.with(Serdes.String(), Serdes.String()));
+            final KStream<String, String> otherStream = builder.stream(Pattern.compile("not-a-match"));
+
+            pattern1Stream
+                .selectKey((k, v) -> k)
+                .groupByKey()
+                .aggregate(() -> "", (k, v, a) -> v)
+                .toStream().to(outputTopic, Produced.with(Serdes.String(), Serdes.String()));
```

Comment:
I would even add `otherStream` like:
```suggestion
                .toStream()
                .merge(otherStream)
                .to(outputTopic, Produced.with(Serdes.String(), Serdes.String()));
```

Reply:
But if we merge then won't that merge the subtopologies as well? (We would still have two subtopologies due to the upstream key-changing operation/repartition, but I wanted the test to cover different "kinds" of subtopologies like this)

Reply:
You would have a source node in one sub-topology an a source node in the other topology. I thought that was the pattern in the bug report, but I now realized that the bug report uses a pattern similar to the one you did specified. Wouldn't it make sense to test both?

Reply:
There are two ways to have more than one source node: either reading from a different input topic/pattern, or via a repartition. I agree that we should test both of these cases, but I'd prefer to do so in a single integration test rather than in two separate integration tests, to avoid making the test suite even longer